### DISCUSSION TITLE
Project Localization (Translation)

### DISCUSSION THREAD

**Discussion by ChanoSan** [comit-network/xmr-btc-swap#679] [Ideas]
Are there any plans to localize the swap tool (and/or the asb tool) into other languages via something like Transifex or Weblate?
I ask because I've been translating SethForPrivacy's introductory guide to using COMIT's swap tool, and I'd like to assist in localizing the tool itself, if that's in the cards.

Thanks

**Comment by thomaseizinger** [comment]
I have no experience in how that would work in practice for a CLI like the swap tool or the ASB. Generally open to it although I am not gonna be able to do any of the coding myself.

Could you work out a proposal on what we would have to do technically?

**Reply by ChanoSan** [reply]
> Could you work out a proposal on what we would have to do technically?

I'm normally only working on the translator's side, but let me consult with some people on the project management side of some other projects and I'll get back to you here.

**Reply by ChanoSan** [reply]
I did some reading and asking around, and it turns out the process is a little more involved than I realized. According to Transifex's [guide](https://www.transifex.com/blog/2015/software-localization-guide/) on project localization:

> "For a software developer, i18n means designing a localizable user-interface and abstracting all localizable elements (user-visible strings, locale-specific data like date, time, and currency formats, keyboard usage, etc.) out of your application source code and into external files that can be made available for translations"

From what I understand, this would entail creating an external file containing user-visible strings (possibly called on by the source code depending on the target language of the binary its creating) which could be integrated with Transifex. They have a guide on supported formats [here](https://docs.transifex.com/formats/introduction). 

Transifex pulls the strings from Github and pipes them through the translation system, then pushes the results of finished translations back to Github in the form of a new language file. So if the original strings file was "strings_EN.txt", it would return Japanese translations to "strings_JA.txt", German to "strings_DE.txt" etc. I believe the specifics of exactly how translations are pushed back is configurable on Transifex's end.

Unfortunately, I'm forced to speculate on a great deal because I've never worked that side of things. It is a great deal more work than I realized, though, and I imagine you're busy enough right now. Perhaps this is best left as a fevered pipe-dream until somebody with the right skills and available time comes along to attempt it.

**Reply by thomaseizinger** [reply]
Thanks for doing this research! It is good to capture this knowledge here. I am afraid we won't get around to enable this in quite a while.

**Reply by ChanoSan** [reply]
Understood. I guess the user-facing strings in CLI swap tool aren't so difficult as to be unmanageable, a translated guide should be sufficient in the short term. I imagine the BTC -> XMR route will be of more interest to the Japanese crypto community than the inverse right now anyway.

Thanks again.
